src/S_Object/S_Capsule.cpp

```diff
@@ -0,0 +1,30 @@
+#include <sch/S_Object/S_Capsule.h>
+
+using namespace sch;
+S_Capsule::S_Capsule(Point3 p1, Point3 p2, Scalar radius): p1_(p1),p2_(p2),radius_(radius)
+{
+}
+
+S_Capsule::~S_Capsule()
+{
+}
+
+Point3 S_Capsule::l_Support(const Vector3& v, int& /*lastFeature*/)const
+{
+  Point3 d(p2_);
+  d-=p1_;
+
+  if (d*v>0)
+    d=p2_;
```

Comment:
Braces around single conditional statements are better imho (also, from a quick glimpse around the code, it seems to be the norm in the rest of `sch-core`)